lua/ttt2/libraries/playermodels.lua

```diff
@@ -0,0 +1,190 @@
+
+		local playerModelPoolNames = {}
+
+		for i = 1, #playermodelPool do
+			if playermodelPool[i].name == playermodels.initializedFlag then continue end
+
+			playerModelPoolNames[#playerModelPoolNames + 1] = playermodelPool[i].name
+		end
+
+		return playerModelPoolNames
+	else
+		return playermodels.selectedModels or {}
+	end
+end
+
+---
+-- Checks if a provided model is in the selection pool.
+-- @note While this function is shared, the data is only available for superadmin on the
+-- client, if no manual sync is triggered.
+-- @param string name The name of the model
+-- @return boolean Returns true, if the model is in the selection pool
+-- @realm shared
+function playermodels.HasSelectedModel(name)
+	local models = playermodels.GetSelectedModels()
+
+	for i = 1, #models do
+		if models[i] ~= name then continue end
+
+		return true
+	end
+
+	return false
+end
+
+
+if CLIENT then
+	net.ReceiveStream("TTT2StreamModelTable", function(data)
+		playermodels.selectedModels = data
+	end)
+end
+
+if SERVER then
+	---
+	-- @realm server
+	local cvCustomModels = CreateConVar("ttt2_use_custom_models", "0", {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+
+	local initialModels = {
+		["css_phoenix"] = true,
+		["css_arctic"] = true,
+		["css_guerilla"] = true,
+		["css_leet"] = true
+	}
+
+	-- extend the playermodels scope on the server
+	playermodels.sqltable = "ttt2_playermodel_pool"
+	playermodels.savingKeys = {
+		state = {typ = "bool", default = false}
+	}
+	playermodels.initializedFlag = "__INITIALIZED__"
+
+	playermodels.fallbackModel = "models/player/phoenix.mdl"
+
+	---
+	-- Initializes the database. This adds all models and sets the default models to true, if it
+	-- is the first init on this server.
+	-- @return boolean Returns false, if it failed during the process
+	-- @internal
+	-- @realm server
+	function playermodels.InitializeDatabase()
+		if not sql.CreateSqlTable(playermodels.sqltable, playermodels.savingKeys) then
+			return false
+		end
+
+		local playermodelPoolModel = orm.Make(playermodels.sqltable)
+		local initializedFlag = playermodelPoolModel:Find(playermodels.initializedFlag)
+		local isNewTable = initializedFlag and (initializedFlag.state ~= true) or true
+
+		for name in pairs(playerManagerAllValidModels()) do
+			if playermodelPoolModel:Find(name) then continue end
+
+			playermodelPoolModel:New({name = name, state = (isNewTable and initialModels[name]) or false}):Save()
+		end
+
+		-- mark table as initialized
+		playermodelPoolModel:New({name = playermodels.initializedFlag, state = true}):Save()
+
+		return true
+	end
+
+	---
+	-- Streams the playermodel selection pool to clients. By default the clients are all superadmin players.
+	-- @param[opt] table|Player plys The players that should receive the update, all superadmins if nil
+	-- @realm server
+	function playermodels.StreamToSelectedClients(plys)
+		plys = plys or util.GetFilteredPlayers(function(ply)
+			return ply:IsSuperAdmin()
```

Comment:
Clients that become superadmin later won't have this table streamed to them initially

Reply:
I see, yes. I think this can be simplified

Reply:
There currently is no way of detecting a change of the usergroup. Overwriting the function in TTT2 would not solve it, because most admin mods overwrite the function again: https://github.com/Facepunch/garrysmod-requests/issues/1139

Therefore I think it is alright to keep it that way. The only drawback is that ne new superadmin can't see selected models. As soon as he clicks one, all models are stramed to him. In my opinion this is an edge case that can be accepted.